src/xpub/signer_config.rs

```diff
@@ -0,0 +1,126 @@
+use super::signing_client::*;
+use crate::error::*;
+use chacha20poly1305::{
+    aead::{Aead, AeadCore, KeyInit, OsRng},
+    ChaCha20Poly1305,
+};
+use serde::{Deserialize, Serialize};
+
+pub type EncryptionKey = chacha20poly1305::Key;
+pub(super) struct ConfigCyper(pub(super) Vec<u8>);
+pub(super) struct Nonce(pub(super) Vec<u8>);
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Default)]
+pub struct SignerEncryptionConfig {
+    #[serde(with = "key_serialization")]
+    pub key: Option<EncryptionKey>,
+}
+
+mod key_serialization {
+    use super::EncryptionKey;
+    use serde::de::{Error, Unexpected};
+    use serde::{Deserialize, Deserializer, Serializer};
+
+    pub(super) fn deserialize<'de, D>(deserializer: D) -> Result<Option<EncryptionKey>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let hex_string = String::deserialize(deserializer)?;
+        let key_vec = hex::decode(&hex_string).map_err(|_err| {
+            D::Error::invalid_value(Unexpected::Str(&hex_string), &"valid hex string")
+        })?;
+
+        let key_bytes = key_vec.as_slice();
+        if key_bytes.len() == 32 {
+            Ok(Some(*chacha20poly1305::Key::from_slice(key_bytes)))
+        } else {
+            Err(D::Error::invalid_length(key_bytes.len(), &"32-byte array"))
+        }
+    }
+
+    pub(super) fn serialize<S>(
+        key: &Option<EncryptionKey>,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match key {
+            Some(key) => {
+                let key_bytes = key.as_slice();
+                let hex_string = hex::encode(key_bytes);
+                serializer.serialize_str(&hex_string)
+            }
+            None => serializer.serialize_none(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum SignerConfig {
+    Lnd(LndSignerConfig),
+    Bitcoind(BitcoindSignerConfig),
+}
+
+impl SignerConfig {
+    pub(super) fn encrypt(&self, key: &EncryptionKey) -> Result<(ConfigCyper, Nonce), BriaError> {
+        let cipher = ChaCha20Poly1305::new(key);
+        let nonce = ChaCha20Poly1305::generate_nonce(&mut OsRng);
+        let encrypted_config = cipher
+            .encrypt(&nonce, serde_json::to_vec(self)?.as_slice())
+            .unwrap();
+
+        Ok((ConfigCyper(encrypted_config), Nonce(nonce.to_vec())))
+    }
+
+    pub(super) fn _decrypt(
```

Comment:
why `_decrypt`?

Reply:
just to pass make check-code. Will remove this.